.github/workflows/pr-staging-to-prod.yml

```diff
@@ -0,0 +1,64 @@
+name: 3 | Generate PR for releases/staging to releases/prod
+
+# This CI job is responsible for generating PRs that bring the HEAD of `releases/staging` into `releases/prod`.
+# These PRs are meant to be the only (standard) way that code is merged into the `releases/prod` branch.
+#
+# https://github.com/peter-evans/create-pull-request/blob/main/docs/examples.md#keep-a-branch-up-to-date-with-another
+
+on:
+  workflow_dispatch:
+
+jobs:
+  prod-gen-pr:
+    name: 'Generate PR for merging to releases/prod branch'
+    runs-on: ubuntu-latest
+    environment:
+      name: release
+    steps:
+      - uses: actions/checkout@8e5e7e5ab8b370d6c329ec480221332ada57f0ab
+        with:
+          token: ${{ secrets.RELEASE_SERVICE_ACCESS_TOKEN }}
+          ref: main
+      - uses: actions/checkout@8e5e7e5ab8b370d6c329ec480221332ada57f0ab
+        with:
+          token: ${{ secrets.RELEASE_SERVICE_ACCESS_TOKEN }}
+          ref: releases/prod
+      - name: Reset promotion branch
+        run: |
+          git fetch origin releases/staging:releases/staging
+          git reset --hard releases/staging
+      - name: Setup git
+        run: |
+          git config user.name "UL Mobile Service Account"
+          git config user.email "hello-happy-puppy@users.noreply.github.com"
+      - name: Add CODEOWNERS file
+        run: |
+          echo "@uniswap/web-reviewers" > CODEOWNERS
+          git add CODEOWNERS
+          git commit -m "ci: add CODEOWNERS file"
```

Comment:
Couldn't we just add this file to the repo, and not have to do it as part of a CI step?

Reply:
The file needs to be added only for a single branch unfortunately :( If we added this to `main` then the `@uniswap/web-reviewers` group would be required to approve every PR targeting `main`.

Not a lot of good documentation on it but I found this instructive - https://stackoverflow.com/questions/75380368/github-codeowners-for-each-branch

Reply:
We already require that `@Uniswap/web-reviewers` approve every PR targeting `main`, so I think that would be fine. They own the codebase, so they should make all of the PR _and_ deploy approvals.

We can also segment off sub-sections with nested CODEOWNERS